### PR TITLE
Example plugin for a perpective including onboarding

### PR DESCRIPTION
**Pull request by marcushoepfner**
Issue #633

Shows the usage of onboarding text, image and shortcuts.

### CODE REVIEW

**Review by BeckerWdf — CHANGES_REQUESTED** [review]
Pls. add a proper version of the icon (incl. a @2x.png).
Ideally take the SVG version of the icon. Resize that and create the two PNGs out of it.

**Comment by marcushoepfner** [comment]
> @marcushoepfner If I use interace rebase on this commit onto the PR for the onboarding and import the modified projects I see setOnboarding called on the StackRender but the empty editor area is still empty in a runtime Eclipse.
> 
> Anything else I need to do to see the change?

You need to switch to "Example Perspective" perspective.

**Comment by vogella** [comment]
> > @marcushoepfner If I use interace rebase on this commit onto the PR for the onboarding and import the modified projects I see setOnboarding called on the StackRender but the empty editor area is still empty in a runtime Eclipse.
> > Anything else I need to do to see the change?
> 
> You need to switch to "Example Perspective" perspective.

Of course. :-) I had one of the old plug-ins from the target perspective in my runtime IDE. Works better now.

**Comment by marcushoepfner** [comment]
> Pls. add a proper version of the icon (incl. a @2x.png). Ideally take the SVG version of the icon. Resize that and create the two PNGs out of it.

done

**Comment by marcushoepfner** [comment]
I changed the onboarding text to something like IntelliJ uses:
`Open a file or drop files here to open them.`
Before I had `Welcome to the example perspective.` But I don't want this as an example. At the end every perspective has a text like `Welcome to my ...`, which does not any value. If in doubt, I would even recommend a perspective, don't add an onboarding text.

**Comment by BeckerWdf** [comment]
> I changed the onboarding text to something like IntelliJ uses: `Open a file or drop files here to open them.` Before I had `Welcome to the example perspective.` But I don't want this as an example. At the end every perspective has a text like `Welcome to my ...`, which does not any value. If in doubt, I would even recommend a perspective, don't add an onboarding text.

I absolutely agree with that.

File: examples/org.eclipse.ui.examples.perspective/plugin.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            icon="$nl$/icons/full/etool16/default_persp.png"
+            id="org.eclipse.ui.examples.perspective.perspective"
+            name="Example Perspective">
+      </perspective>
+   </extension>
+   <extension
+         point="org.eclipse.ui.perspectiveExtensions">
+      <perspectiveExtension
+            editorOnboardingImage="$nl$/icons/default_persp_grey.png"
+            editorOnboardingText="Open a file or drop files here to open them."
```
Comment (BeckerWdf) [inline examples/org.eclipse.ui.examples.perspective/plugin.xml]: This should be externalized.

File: examples/org.eclipse.ui.examples.perspective/icons/default_persp_grey.png
Comment (BeckerWdf) [inline examples/org.eclipse.ui.examples.perspective/icons/default_persp_grey.png:1]: i think these should go into the "full" folder

File: examples/org.eclipse.ui.examples.perspective/icons/default_persp_grey@2x.png
Comment (BeckerWdf) [inline examples/org.eclipse.ui.examples.perspective/icons/default_persp_grey@2x.png:1]: We should remove the empty border

**Comment by vogella** [comment]
@marcushoepfner can you update this one also we that we can merge the example?

**Comment by marcushoepfner** [comment]
> @marcushoepfner can you update this one also we that we can merge the example?

working on it, wait a few minutes pls :)

File: examples/org.eclipse.ui.examples.perspective/plugin.xml (hunk 1 above)
Comment (marcushoepfner) [reply examples/org.eclipse.ui.examples.perspective/plugin.xml]: I have added them to plugin.properties like e.g. plugin o.e.ui does it with extension texts.
Unfortunately they are not correctly retrieved in target and hence not shown in ui.
When I use the externalize wizard it adds them to a new file bundle.properties.

What am I doing wrong here?

Comment (marcushoepfner) [reply examples/org.eclipse.ui.examples.perspective/plugin.xml]: Ah found it...
`Bundle-Localization: plugin` need to be added to Manifest.MF.

Now it works.

**Review by BeckerWdf — APPROVED** [review]

**Pull request merged**